Unstructured-IO/unstructured-inference
# feat: add logging config

Adds a proper `logger_config.yaml` for uvicorn.

**Test Instructions** 

`make run-app-dev` should include formatting output specific to `logger_config.yaml` as shown below.

```
(uninf) MacBook-Pro-2:unstructured-inference cragwolfe$ make run-app-dev
PYTHONPATH=. uvicorn unstructured_inference.api:app --log-config logger_config.yaml --reload
2023-03-14 23:06:17,342 uvicorn.error INFO Will watch for changes in these directories: ['/Users/cragwolfe/r/unstructured-inference']
2023-03-14 23:06:17,342 uvicorn.error INFO Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
2023-03-14 23:06:17,342 uvicorn.error INFO Started reloader process [32697] using WatchFiles
2023-03-14 23:06:19,721 uvicorn.error INFO Started server process [32708]
2023-03-14 23:06:19,722 uvicorn.error INFO Waiting for application startup.
2023-03-14 23:06:19,722 uvicorn.error INFO Application startup complete.
```

NOTE: it is mildly awkward to read `uvicorn.error` as the name of the logger that uvicorn writes too when an actual log message may be at INFO, but it's good to know which logger it is coming from!

Test hitting the API with:

```
curl -X 'POST' 'http://localhost:8000/layout/default/pdf' \
-F 'file=@<your-pdf>.pdf' \
-F include_elems=Text \
-F include_elems=Title \
 | jq .
```
And note the DEBUG output line from the `unstructured_inference` logger:

```
...
2023-03-14 23:09:57,087 unstructured_inference DEBUG Running OCR on text block ...
...
```
